### ISSUE TITLE
PHP curl extension not work

### ISSUE THREAD

**Issue by julienallexandre** [oerdnj/deb.sury.org#1705]
**Frequently asked questions**

- [x] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**

**The problem is described in Additional context**

A connection timeout was encountered. If you intend to run Composer without connecting to the internet, run the command again prefixed with COMPOSER_DISABLE_NETWORK=1 to make Composer run in offline mode.

**To Reproduce**

`composer update / composer install / composer`

**Distribution :**

 - OS:  Windows 11 21H2 22000.376 with WSL Ubuntu 20.04.3 LTS
 - Architecture: amd64
 - Repository: ppa:ondrej/php
 - Utils : Composer 2.2.4 with PHP 8.1.1

**Package(s) :**

php8.1-curl:
  Installed: 8.1.1-2+ubuntu20.04.1+deb.sury.org+1
  Candidate: 8.1.1-2+ubuntu20.04.1+deb.sury.org+1
  Version table:
 *** 8.1.1-2+ubuntu20.04.1+deb.sury.org+1 500
        500 http://ppa.launchpad.net/ondrej/php/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status

**Additional context**

I had opened an issue with composer believing that it was coming from them but no I realize that it is indeed the curl extension of php that causes this problem, I give you the link concerning the problem if you can help me to fix it thanks 
Link : [https://github.com/composer/composer/issues/10445](https://github.com/composer/composer/issues/10445)

**Comment by rfay** [comment]
For what it's worth, I'm not seeing any issues with composer 2.2.4 and php8.1.1. [ddev](https://github.com/drud/ddev) uses this for everything (inside docker containers) on both amd64 and arm64, and tests on WSL2 and lots of other places. 

You say in the other issue that you also have this problem with php7.4, right?

I think it's more likely that you have some strange firewall or vpn issue, please make sure to disable all of those and use on a different network. Also, have you tried it on another computer (and a simpler network)?

```bash
gitpod@t3simple-web:/var/www/html$ composer --version
Composer version 2.2.4 2022-01-08 12:30:42
gitpod@t3simple-web:/var/www/html$ php --version
PHP 8.1.1 (cli) (built: Dec 20 2021 21:35:13) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.1, Copyright (c), by Zend Technologies
    with blackfire v1.73.0~linux-x64-non_zts81, https://blackfire.io, by Blackfire
gitpod@t3simple-web:/var/www/html$ dpkg -l php8.1-curl
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                                    Architecture Description
+++-==============-==========================================-============-=================================
ii  php8.1-curl    8.1.1-1+0~20211220.11+debian11~1.gbp519568 amd64        CURL module for PHP
gitpod@t3simple-web:/var/www/html$ composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
Generating autoload files
> typo3cms install:fixfolderstructure
No action performed!
46 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
gitpod@t3simple-web:/var/www/html$ 
```

**Comment by oerdnj** [comment]
One way or another, this is not a packaging issue, so I don't know how could I help you.

**Comment by julienallexandre** [comment]
I have tested with another computer and I don't have a vpn. I have even deactivated the firewall and it doesn't change anything. Yes, I have the same problem with 7.4 since the last update (apt update php) before that it worked well and the problem only occurs when I have the php curl extension on the machine

**Comment by julienallexandre** [comment]
@rfay You are on the (built) version that does not bug how can I get back to this one? 

On her (built: Dec 20 2021 21:35:13) I had no problem since I am on her (built: Dec 31 2021 07:26:20) I have the problem  

```
julien@MNKY-PC:~/Code$ composer --version
Composer version 2.2.4 2022-01-08 12:30:42
julien@MNKY-PC:~/Code$ php --version
PHP 8.1.1 (cli) (built: Dec 31 2021 07:26:20) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.1, Copyright (c), by Zend Technologies
```

**Comment by rfay** [comment]
I guess you must be using the Ubuntu ppa as you say above, but on Debian 11 Bullseye, these are the latest:

```bash
$ dpkg -l php8.1-cli
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                                    Architecture Description
+++-==============-==========================================-============-=======================================================
ii  php8.1-cli     8.1.1-1+0~20211220.11+debian11~1.gbp519568 amd64        command-line interpreter for the PHP scripting language
gitpod@t3simple-web:/var/www/html$ dpkg -l php8.1-curl
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                                    Architecture Description
+++-==============-==========================================-============-=================================
ii  php8.1-curl    8.1.1-1+0~20211220.11+debian11~1.gbp519568 amd64        CURL module for PHP
gitpod@t3simple-web:/var/www/html$ php --version
PHP 8.1.1 (cli) (built: Dec 20 2021 21:35:13) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.1, Copyright (c), by Zend Technologies
```

**Comment by rfay** [comment]
On Ubuntu 20.04 amd64 I had no trouble with these versions and composer:
```
l# php --version
PHP 8.1.1 (cli) (built: Dec 31 2021 07:26:20) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.1, Copyright (c), by Zend Technologies
root@bac232c2c809:/var/www/html# composer --version
Composer version 2.2.4 2022-01-08 12:30:42
root@bac232c2c809:/var/www/html# dpkg -l php8.1-cli  
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                              Architecture Description
+++-==============-====================================-============-=======================================================
ii  php8.1-cli     8.1.1-2+ubuntu20.04.1+deb.sury.org+1 amd64        command-line interpreter for the PHP scripting language
root@bac232c2c809:/var/www/html# dpkg -l php8.1-curl
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                              Architecture Description
+++-==============-====================================-============-=================================
ii  php8.1-curl    8.1.1-2+ubuntu20.04.1+deb.sury.org+1 amd64        CURL module for PHP
```


One thing you might want to check is whether you have latest libcurl and such. I imagine you've already done a `sudo apt-get update && sudo apt-get upgrade` a few times, so that should be taken care of.

**Comment by rfay** [comment]
You can run a docker container to experiment with this. That's what I did:

```
docker run -it --rm ubuntu:20.04
apt-get update && apt install software-properties-common
add-apt-repository ppa:ondrej/php
apt-get update
apt-get install -y php8.1-cli php8.1-curl php8.1-gd php8.1-dom php8.1-zip
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php -r "if (hash_file('sha384', 'composer-setup.php') === '906a84df04cea2aa72f40b5f787e49f22d4c2f19492ac310e8cba5b96ac8b64115ac402c8cd292b8a03482574915d1a8') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
php composer-setup.php
php -r "unlink('composer-setup.php');"
mv composer.phar /usr/local/bin/composer
mkdir junk && cd junk
composer create-project drupal/recommended-project .

**Comment by julienallexandre** [comment]
I'm sorry to have made you lose time. But I found the source of the problem and I'm very annoyed that it took me 3 days to find it. I explain myself I have 2 pc windows 11 and on both I use wsl but I have bitdefender and malwaresbytes installed, I had deactivated them but nothing changed and after a long reflection I decided to reinstall one of the 2 and bingo it works. I did some step by step: 1) malwaresbytes no problem 2) bitdefender PROBLEM! and yes this bitdefender must have done a crappy update on windows 11 that causes me this problem but I can't find it and in the end it's their shitty firewall that needs to be deactivated so I can keep the antivirus but not the firewall I even tested that [https://community.bitdefender.com/en/discussion/82809/bitdefender-firewall-wont-allow-wsl-traffic](https://community.bitdefender.com/en/discussion/82809/bitdefender-firewall-wont-allow-wsl-traffic) but it didn't change anything because I had the connection in short, all that to thank you for taking the time to answer me, next time I won't be looking for 3 days I'll be thinking about this shitty firewall, in short, a happy new year to you and once again, thank you
